src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -362,8 +363,14 @@ function fetchAvailabilityData(global) {
   if (!availabilityData) {
     let f = global.resource('availability-data');
     // NOTE: This has to be sync, so we can use this data in if conditions.
-    let rawData = fs.readFileSync(f, { encoding: 'utf8', flag: 'r' });
-    availabilityData = YAML.parse(rawData);
+    if (f) {
+      let rawData = fs.readFileSync(f, { encoding: 'utf8', flag: 'r' });
+      availabilityData = YAML.parse(rawData);
+    } else {
+      throw new Error(
+        `Resource availability-data not found among the context resources. Check your template.json file.`
```

Comment:
Is this just because the exception is nicer than what `readFileSync` throws if `f` is None?

Reply:
I was thinking of catching it before it even goes ahead anymore.